StreamChatClient_v3/Database/DatabaseSession.swift

```diff
@@ -73,5 +73,14 @@ extension DatabaseSession {
                 currentUserDTO.unreadMessagesCount = Int16(unreadCount.messages)
             }
         }
+        
+        // Save message data (must be always done afther the channel data!)
+        if let message = payload.message {
+            if let cid = payload.cid {
+                try saveMessage(payload: message, for: cid)
+            } else {
+                log.error("Message payload \(message) can't be saved because `cid` is missing. Ignoring.")
+            }
```

Comment:
Now you can get untapped value from Payload with func `value()`
```suggestion
            let cid = try payload.value(at: \.cid)
            try saveMessage(payload: message, for: cid)
```

Reply:
Interesting, thanks. I think the code you suggest has slightly different behavior so I'm gonna stick with the original implementation. However, good to know we have this in our toolbox 👍